migrations/dimensions-config/1fae088c80b6.py

```diff
@@ -0,0 +1,323 @@
+
+    # Create index on instrument
+    _LOG.info("Creating instrument index for day_obs table")
+    op.create_index(
+        "day_obs_fkidx_instrument",
+        "day_obs",
+        ["instrument"],
+        schema=ctx.schema,
+    )
+
+    # Update exposure table to reference day_obs table
+    _LOG.info("Updating exposure table to reference day_obs table")
+    with op.batch_alter_table("exposure", schema=ctx.schema) as batch_op:
+        batch_op.alter_column("day_obs", nullable=False)
+        batch_op.create_foreign_key(
+            constraint_name="fkey_exposure_day_obs_instrument_id_instrument_day_obs",
+            referent_table="day_obs",
+            local_cols=["instrument", "day_obs"],
+            remote_cols=["instrument", "id"],
+            referent_schema=ctx.schema,
+        )
+
+    # Create index on exposure for day_obs fkey
+    op.create_index(
+        "exposure_fkidx_instrument_day_obs",
+        "exposure",
+        ["instrument", "day_obs"],
+        schema=ctx.schema,
+    )
+
+
+def _migrate_dimensions_json(ctx: _Context) -> None:
+    _LOG.info("Updating dimensions.json in ButlerAttributes")
+    ctx.attributes.replace_dimensions_json(6)
+
+
+def _generate_day_obs_rows(ctx: _Context) -> Iterator[dict]:
+    exposure_table = ctx.get_table("exposure")
+    select = sa.select(
+        exposure_table.columns["instrument"],
+        exposure_table.columns["day_obs"],
+    ).distinct()
+    rows = ctx.bind.execute(select).all()
+
+    instrument_fetcher = _InstrumentFetcher(ctx)
+    for row in rows:
+        day_obs = row.day_obs
+
+        # Different instruments define the start and end times for day_obs differently.
+        instrument_name = row.instrument
+        instrument_class = instrument_fetcher.get_instrument(instrument_name)
+        offset = _get_day_obs_offset(instrument_name, instrument_class, day_obs)
+
+        timespan = Timespan.from_day_obs(day_obs, offset)
+        yield {
+            "instrument": row.instrument,
+            "id": day_obs,
+            **format_timespan_value(timespan, "timespan", ctx.dialect),
+        }
+
+
+def _get_day_obs_offset(instrument_name: str, instrument: _Instrument, day_obs: int) -> int:
+    day_as_astropy_time = Timespan.from_day_obs(day_obs, 0).begin
+    translator = instrument.translatorClass
+    if translator is None:
+        raise TypeError(
+            f"Instrument {instrument_name} does not have a translatorClass defined,"
+            " cannot determine offset for day_obs."
+        )
+    offset = translator.observing_date_to_offset(day_as_astropy_time)
+    # Convert astropy TimeDelta to integer seconds.
+    return round(offset.to_value("s"))
+
+
+def _make_string_check_constraint(
+    field_name: str, max_length: int, constraint_name: str
+) -> sa.schema.CheckConstraint:
+    return sa.schema.CheckConstraint(
+        f'length("{field_name}")<={max_length} AND length("{field_name}")>=1', name=constraint_name
```

Comment:
I think we already have more than one case where we generate this string, maybe it's time to move it to a common module (`naming` is probably fine).